### PR TITLE
The Pantry should reject non-block sized writes

### PR DESCRIPTION
**Pull request by jmpesp**
Instead of panicking, the Pantry should reject non-block sized writes with a 400 level response. The panic came from Volume, which has been corrected to return a CrucibleError instead.

Unfortunately everything the Pantry returns is currently a 500: this commit also defines different HttpError conversions for CrucibleError, and the Pantry now returns those.

This commit also removes the `impl Into<std::io::Error> for CrucibleError`: the author of that code didn't realize that `impl From` works as long as one of the types is in the current crate!

### CODE REVIEW

File: pantry/src/pantry.rs [hunk 1]
```diff
@@ -171,19 +184,26 @@ impl PantryEntry {
...
             })
-            .await?;
+            .await
+            .map_err(|e| CrucibleError::GenericError(e.to_string()))?;
 
             let content_length = response
                 .headers()
                 .get(reqwest::header::CONTENT_LENGTH)
                 .ok_or("no content length!")
                 .map_err(|e| anyhow!(e))?;
```
Comment (leftwo) [inline pantry/src/pantry.rs:194]: Will this ? turn into a `CrucibleError`?  Do we care if it does not?

File: upstairs/src/volume.rs [hunk 2]
```diff
@@ -481,6 +481,12 @@ impl Volume {
             return Ok(());
         }
 
+        let bs = self.get_block_size().await?;
```
Comment (leftwo) [inline upstairs/src/volume.rs:484]: Do we want the same check for the read path as well?

**Review by leftwo — APPROVED** [review]
Just a few small questions

File: upstairs/src/volume.rs (hunk 2 above)
Comment (jmpesp) [reply upstairs/src/volume.rs:484]: Yes! see 156eb29

File: pantry/src/pantry.rs (hunk 1 above)
Comment (jmpesp) [reply pantry/src/pantry.rs:194]: It will turn into a `CrucibleError::GenericError`, yeah, which will get returned by the Pantry as a 500.

**Pull request merged**